### PR TITLE
.NET - enable bytecode instrumentation

### PR DESCRIPTION
**Pull request by Kielek**
Enables bytecode instrumentations. All of them are enabled by default.
For v.0.3.1-beta.1 it extends set of supported libraries: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/v0.3.1-beta.1/docs/config.md#instrumented-traces-libraries-and-frameworks

### CODE REVIEW

**Comment by Kielek** [comment]
@open-telemetry/dotnet-instrumentation-approvers FYI

**Comment by Kielek** [comment]
@avadhut123pisal, FYI

**Review by lachmatt — APPROVED** [review]

File: pkg/instrumentation/dotnet.go [hunk 1]
```diff
@@ -116,3 +112,25 @@ func injectDotNetSDK(logger logr.Logger, dotNetSpec v1alpha1.DotNet, pod corev1.
...
+		container.Env = append(container.Env, corev1.EnvVar{
+			Name:  envVarName,
+			Value: envVarValue,
+		})
+		return true
+	}
+
+	if container.Env[idx].ValueFrom != nil {
+		// TODO add to status object or submit it as an event
+		logger.Info("Skipping DotNet SDK injection, the container defines env var value via ValueFrom", "envVar", envVarName, "container", container.Name)
```
Comment (pellared) [inline pkg/instrumentation/dotnet.go:128]: I think this behavior and the log message are misleading.

When `false` is returned then it interrupts processing next env vars.
It could leave the container in a "semi-instrumented" state. Is it expected?

This comment applies also to the code BEFORE the changes introduced in this PR. Therefore, it would be better to make a separate PR to address it.

File: pkg/instrumentation/dotnet.go [hunk 2]
```diff
@@ -116,3 +112,25 @@ func injectDotNetSDK(logger logr.Logger, dotNetSpec v1alpha1.DotNet, pod corev1.
...
+		return false
+	}
+	if concatValues {
```
Comment (pellared) [inline pkg/instrumentation/dotnet.go:131]: nit: add a blank line for consistency (and easier reading)

**Review by pellared — APPROVED** [review]

File: pkg/instrumentation/dotnet.go (hunk 2 above)
Comment (Kielek) [reply pkg/instrumentation/dotnet.go:132]: Fixed: 10a7e986e51fde03ae59a29e71da8fb6e25eb6d3

**Review by nrcventura — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]

**Comment by pellared** [comment]
@Kielek I think it MAY be good to offer an easy way to disable byte-code instrumentation (disable attaching the .NET Profiler) as this can have negative consequences on the instrumented application's performance (maybe it was not enabled by default on purpose?)

**Comment by Kielek** [comment]
> @Kielek I think it MAY be good to offer an easy way to disable byte-code instrumentation (disable attaching the .NET Profiler) as this can have negative consequences on the instrumented application's performance (maybe it was not enabled by default on purpose?)

My perspective is to offer as broad auto-instrumentation as possible.
It can be disabled by the user by setting `CORECLR_ENABLE_PROFILING` to `0`.
